File: Services/MedicalReport/src/main/java/org/example/medicalreport/Models/entities/Appointment.java
package org.example.medicalreport.Models.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import org.example.medicalreport.Models.enums.ConsultationType;

import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
@ToString
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    LocalDateTime date;
    String reason;
    @Enumerated(EnumType.STRING)
    ConsultationType consultationType;
    long cabinetId;
    @ToString.Exclude
    @ManyToOne
    private MedicalRecord medicalRecord;

}

File: front/src/app/Models/Appointment.ts
import { Cons } from "rxjs";
import { ConsultationType } from "./enums";
import { MedicalRecord } from "./MedicalRecord";

export class Appointment {
  id?: number;
  date!: Date;
  reason!: string;
  consultationType!: ConsultationType;
  medicalRecord!: MedicalRecord;
  cabinetId!: number;
}